uqcsbot/utils/uq_course_utils.py

```diff
@@ -44,10 +119,30 @@ class ProfileNotFoundException(Exception):
     Raised when a profile cannot be found for a given course.
     """
 
-    def __init__(self, course_name):
-        self.message = f"Could not find profile for course '{course_name}'."
+    def __init__(self, course_name: str, offering: Optional[Offering] = None):
+        if offering is None:
+            self.message = f"Could not find profile for course '{course_name}'. This profile may not be out yet."
+        else:
+            self.message = f"Could not find profile for course '{course_name}' during semester {offering.semester} at {offering.campus} done in mode '{offering.mode}'. This profile may not be out yet."
         self.course_name = course_name
-        super().__init__(self.message, self.course_name)
+        self.offering = offering
+        super().__init__(self.message, self.course_name, self.offering)
+
+
+class AssessmentNotFoundException(Exception):
```

Comment:
Personal preference: exceptions up the top makes them easier to skim-read.

Reply:
The class `Offering` has to be defined for some of the exceptions, so it would be difficult to move the exceptions above it.

Reply:
fair cop; it's a readability preference, so actual limitations of code structure trump that.